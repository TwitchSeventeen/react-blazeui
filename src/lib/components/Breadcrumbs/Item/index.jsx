import React from 'react';
import PropTypes from 'prop-types';
import omitOwnProps from '../../../utils/omitOwnProps';

function Item(props) {
  const { children, active } = props;
  const spreadProps = omitOwnProps(props);
  return (
    <li
      {...spreadProps}
      className="c-breadcrumbs__crumb"
      aria-current={active ? 'page' : null}
    >
      { children }
    </li>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Whether or not this item is active
   */
  active: PropTypes.bool,
};

Item.defaultProps = {
  children: null,
  active: false,
};

export default Item;
