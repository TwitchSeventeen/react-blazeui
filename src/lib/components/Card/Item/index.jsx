import React from 'react';
import PropTypes from 'prop-types';

function Item (props) {
  const { children } = props;
  return (
      <div className="c-card__item">
        { children }
      </div>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Item.defaultProps = {
  children: null,
};

export default Item;