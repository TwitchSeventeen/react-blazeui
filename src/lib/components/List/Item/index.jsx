import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Item(props) {
  const {
    children, className, unstyled, dataName, ...other
  } = props;
  const classes = classNames(
    'c-list__item',
    { 'c-list__item--unstyled': unstyled },
    className,
  );

  return (
    <li
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </li>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines an unstyled list
   */
  unstyled: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Item.defaultProps = {
  children: null,
  className: null,
  unstyled: false,
  dataName: 'list-item',
};

export default Item;
