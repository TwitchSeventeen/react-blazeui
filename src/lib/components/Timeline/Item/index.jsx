import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Item(props) {
  const {
    children, as: Component, className, last, type, loading, left, dataName, ...other
  } = props;
  const classes = classNames(
    'c-timeline-item',
    { 'c-timeline-item--last': last },
    { [`c-timeline-item--${type}`]: type },
    { 'c-timeline-item--loading': loading },
    { 'c-timeline-item--left': left },
    className,
  );

  return (
    <Component
      {...other}
      className={classes}
      data-name={dataName}
    >
      <div className="c-timeline-item__body">
        { children }
      </div>
    </Component>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Identifies the last item in a timeline
   */
  last: PropTypes.bool,
  /**
   * The type of item
   */
  type: PropTypes.oneOf([
    'brand',
    'info',
    'warning',
    'success',
    'error',
  ]),
  /**
   * Defines an item that hasn't happened yet
   */
  loading: PropTypes.bool,
  /**
   * Displays an item on the left side of the Timeline
   */
  left: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Item.defaultProps = {
  children: null,
  className: null,
  last: false,
  type: null,
  loading: false,
  left: false,
  dataName: 'timeline-item',
  as: 'li',
};

export default Item;
