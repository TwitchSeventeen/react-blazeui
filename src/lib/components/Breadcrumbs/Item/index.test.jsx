import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Item from './index';

describe('Breadcrumbs.Item', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Item).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Item>
        <a href="/lorem">lorem</a>
      </Item>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly an active Item', () => {
    const { asFragment } = render(
      <Item active>
        lorem
      </Item>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
