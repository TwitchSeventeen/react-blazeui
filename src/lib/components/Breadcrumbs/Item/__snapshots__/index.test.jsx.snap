// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs.Item renders correctly 1`] = `
<li
  aria-current={null}
  className="c-breadcrumbs__crumb"
>
  <a
    href="/lorem"
  >
    lorem
  </a>
</li>
`;

exports[`Breadcrumbs.Item renders correctly an active Item 1`] = `
<li
  aria-current="page"
  className="c-breadcrumbs__crumb"
>
  lorem
</li>
`;
